sammba/registration/struct.py

```diff
@@ -387,130 +418,85 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
     # A weight mask that extends beyond the brain, incorporating some
     # surrounding tissue, is needed to help better define the brain head
     # boundary.
-    out_mask_tool = mask_tool(in_file=out_tcat.outputs.out_file, count=True,
-                              outputtype='NIFTI_GZ')
-    out_mask_tool = mask_tool(in_file=out_tcat.outputs.out_file, union=True,
-                              outputtype='NIFTI_GZ')
-    out_mask_tool = mask_tool(in_file=out_mask_tool.outputs.out_file,
-                              dilate_inputs='4',
-                              outputtype='NIFTI_GZ')
+    if nonlinear_weight is None:
+        out_mask_tool = mask_tool(in_file=out_tcat.outputs.out_file, union=True,
+                                  outputtype='NIFTI_GZ')
+        out_mask_tool = mask_tool(in_file=out_mask_tool.outputs.out_file,
+                                  dilate_inputs='4',
+                                  outputtype='NIFTI_GZ')
+        nonlinear_weight = out_mask_tool.outputs.out_file
 
     ###########################################################################
-    # The input source images are initially transformed prior to registration,
-    # to ensure that they are already quite well-aligned to the template.
-    # To save time, we only achieve one refinement level per step
+    # Description to fill
+    # 
+    # 
     if nonlinear_levels is None:
         nonlinear_levels = [1, 2, 3]
-
-    warped_files = []
-    warp_files = []
-    for affine_transform_file, centered_head_file in zip(
-            affine_transform_files, centered_head_files):
-        out_qwarp = qwarp(
-            in_file=centered_head_file,
-            base_file=out_tstat_allineated_head.outputs.out_file,
-            nmi=True,
-            noneg=True,
-            iwarp=True,
-            weight=out_mask_tool.outputs.out_file,
-            iniwarp=[affine_transform_file],
-            inilev=0,
-            maxlev=nonlinear_levels[0],
-            out_file=fname_presuffix(centered_head_file, suffix='_warped1'))
-        warp_files.append(out_qwarp.outputs.source_warp)
-        warped_files.append(out_qwarp.outputs.warped_source)
-
-    out_tcat = tcat(
-        in_files=warped_files,
-        out_file=os.path.join(write_dir, 'warped_1iter_heads.nii.gz'))
-    out_tstat_warp_head = tstat(in_file=out_tcat.outputs.out_file,
-                                outputtype='NIFTI_GZ')
-
-    ###########################################################################
-    # Then iterative registration from a given level to another is achieved.
-    # Note that any level below a patch size of 25 will not be done (see
-    # 3dQwarp help for further detail).
-    # The input transform is the former warp and needs to be concatenated to
```

Comment:
I think you removed this step and no more concatenate the first warps to `IDENT`. Is it because you figured out what was the bug here and solved it ?

Reply:
I'm in two minds abut this. I didn't note which files generated this bug. But I have not seen it since so am not entirely sure if it ever really was a real bug. Will readd the IDENT NwarpCat anyway as it's only a line and doesn't do any harm